compiler.cpp

```diff
@@ -402,14 +402,13 @@ struct Compilation {
     ~Compilation() = default;
 
     double Cost(const CostPair& pair, const Node& node) {
-        return node->ScriptSize() + Mul(p, pair.sat) + Mul(q, pair.nsat);
+        return node->GetScriptSize() + Mul(p, pair.sat) + Mul(q, pair.nsat);
     }
 
     void Add(const CostPair& pair, Node node) {
         auto new_typ = node->GetType();
         double cost = Cost(pair, node);
-        if (!node->CheckOpsLimit()) return;
-        if (node->GetStackSize() > MAX_STANDARD_P2WSH_STACK_ITEMS) return;
+        if (!node->IsSafeTopLevel()) return;
```

Comment:
I don't think this is right; it will reject all non-B expressions.

Reply:
We could have both a `IsSafe` and `IsSafeTopLevel` and call the former here?
```cpp
bool IsSafe() const { return CheckOpsLimit() && CheckStackSize() && CheckTimeLocksMix() && CheckScriptSize(); }

bool IsSafeTopLevel() const { return GetType() << "Bms"_mst && IsSafe(); }
```

Reply:
I think "ms" should be required for `IsSafe`, not just `IsSafeTopLevel`.